elytron/WFCORE-4360-CredentialStore_Expression_Resolution.adoc

```diff
@@ -0,0 +1,420 @@
+reversibly encrypted attribute expression values.  Other than the support within the expression resolver to decrypt values at runtime we will not be providing
+any tooling to decrypt the previously encrypted tokens.
+
+Automatic encryption of attribute values will not be supported via this enhancement, as multiple steps are required that would be better performed within
+enhancements to the management tooling - each of which would require special consideration based on their own user interfaces.  A big risk when using multiple
+steps is if intermediate representations of the management model are persisted as these intermediate representations could contain the clear text values.
+
+As with other CredentialStore use cases no automatic replication of the store or it's entries are supported with this enhancement.
+
+This enhancement will not add support for migrating expressions to a different credential, however if support for multiple expression resolvers is added at a
+later point there may be opportunities to support migration.
+
+This enhancement is only in relation to expression resolution within the application server's management model - this does not extend to any other descriptors 
+or configuration files.
+
+Expression resolution will only be supported against attributes that already support expression resolution, this enhancement will not perform a review of which
+attributes support expression resolution and will not be changing any attributes to support expression resolution.
+
+Custom expression resolver implementations are outside of the scope of this RFE, adding custom implementations which potentially reference a credential store
+could be a future enhancement.
+
+Resolved expressions are a deliberate decision to move values from the model to an alternative expression resolver, it is not possible to determine the
+expression resolution capabilities of a slave.  Where credential store backed expression resolution is in place transformers will not reject sending those
+expressions to the slave.  However as the `expression=encryption` resource is not supported on the slave that will by itself fail transformation and be
+rejected.
+
+
+== Test Plan
+
+Any utilities added to the Wildfly Elytron project will also be accompanied with their own unit tests.
+
+The major testing of this enhancement will happen within the WildFly Core project allowing us to test in the same location it is implemented.
+
+== Community Documentation
+
+The community documentation will be enhanced to include details of the new feature.
+
+The documentation must sufficiently describe the encryption process to a level which would allow users and third parties to create their own tooling to generate the expressions.  They may wish to accomplish this using Java however they may also choose to do so in alternative languages provided the required cipher algorithms are available.
+
```

Comment:
[Release Note Content](https://github.com/wildfly/wildfly-proposals/blob/master/design-doc-template.adoc#release-note-content) is missing.

Reply:
I will add this one but this is just for Brian for community content, this is not part of the RFE process.

Reply:
Added.

Reply:
Thank you, this can be resolved.